sdk/core/Azure.Core/api/Azure.Core.net461.cs

```diff
@@ -211,6 +225,8 @@ public partial class RequestFailedException : System.Exception, System.Runtime.S
         public RequestFailedException(string message, System.Exception? innerException) { }
         public string? ErrorCode { get { throw null; } }
         public int Status { get { throw null; } }
+        public static Azure.RequestFailedException Create(Azure.Response response, Azure.ResponseError? error = null, System.Exception? innerException = null) { throw null; }
+        public static System.Threading.Tasks.ValueTask<Azure.RequestFailedException> CreateAsync(Azure.Response response, Azure.ResponseError? error = null, System.Exception? innerException = null, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
```

Comment:
Curious: Did we add this factory method to handle the case where the response is not buffered, because there is no way to add an async constructor?

For the sync one, is that to be symmetric with the async factory method?  Because otherwise it might make sense to add a third constructor that takes Azure.ResponseError? to maintain the constructor pattern we've already established.

Reply:
No - it's because the exception message is readonly, so we need to process the response before constructing the exception.

Reply:
Why not process the response in the constructor, like we did for the other constructors?

Reply:
The constructor needs to ultimately call the base constructor on `Exception`, and we need to manipulate all the inputs based on the response. We could make it work, but I believe it would involve duplicating work parsing the response. Secondly, to your original question, we'd need to do async work as well.

Reply:
What I don't understand is that we're already parsing the response in two constructors on this type.  Why would we add a new creation pattern instead of following the existing pattern?

Reply:
It looks like in the [existing code](https://github.com/Azure/azure-sdk-for-net/pull/28466/files#diff-d201011507f12f72bd9d970a544a1f9f1395e535dd55549a6419cb46ecf33b7bR67), we are using the sync path for reading the response content. I suppose it might be worth making this tradeoff if it keep the API cleaner.

Reply:
I wrote up some pros/cons of the various approaches we have discussed [here](https://gist.github.com/christothes/35f276c3b316cb2d4c7b53ee8da85425)

Reply:
We could ensure that error responses are always buffered by adding this into RetryPolicy right before throwing.